File: examples/18_comptime_collections.ts
const PRIME_COUNT: number = 6;

const ZERO_BUFFER_CAPACITY: number = 16;

const HTTP_STATUS_COUNT: number = 4;
function main(): void {
  console.log("=== Compile-time Collections ===");
  console.log("Vec literals:");
  console.log(`  primes: ${PRIME_COUNT} elements`);
  console.log(`  zero buffer: ${ZERO_BUFFER_CAPACITY} elements`);
  console.log("\nHashMap literal via HashMap::from:");
  console.log(`  tracked HTTP statuses: ${HTTP_STATUS_COUNT} entries`);
}

main();

File: examples/18_comptime_collections.rs
pub use std::collections::HashMap;
pub const PRIME_COUNT: i64 = 6;
pub const ZERO_BUFFER_CAPACITY: i64 = 16;
pub const HTTP_STATUS_COUNT: i64 = 4;
pub fn main() -> () {
    println!("=== Compile-time Collections ===");
    println!("Vec literals:");
    println!("  primes: {} elements", PRIME_COUNT);
    println!("  zero buffer: {} elements", ZERO_BUFFER_CAPACITY);
    println!("\nHashMap literal via HashMap::from:");
    println!("  tracked HTTP statuses: {} entries", HTTP_STATUS_COUNT);
}
